src/Cogworks.CogFlare.Core/Services/CachePurgeService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using static Umbraco.Cms.Core.Constants.Conventions;
+
+namespace Cogworks.CogFlare.Core.Services;
+
+public interface ICachePurgeService
+{
+    Task PurgeExternalCacheAsync(IEnumerable<int> ids, CancellationToken cancellationToken, string notificationLabel, bool isMedia = false);
+}
+
+public class CachePurgeService : ICachePurgeService
+{
+    private readonly ICloudFlareCachePurgeService _cloudFlareCachePurgeService;
+    private readonly IUmbracoContentNodeService _umbracoContentNodeService;
+    private readonly IRelationService _relationService;
+    private readonly CogFlareSettings _cogFlareSettings;
+    private readonly ILogger<ExternalCachePurgeComponent> _logger;
+
+    public CachePurgeService(
+        ICloudFlareCachePurgeService cloudFlareCachePurgeService,
+        IUmbracoContentNodeService umbracoContentNodeService,
+        IRelationService relationService,
+        CogFlareSettings cogFlareSettings,
+        ILogger<ExternalCachePurgeComponent> logger)
```

Comment:
Please update the logger to point to current class: `ILogger<CachePurgeService> logger`